x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/operator/lookup/EnrichQuerySourceOperatorTests.java

```diff
@@ -277,14 +248,45 @@ private static DirectoryData makeDirectoryWith(List<List<String>> terms) throws
             for (var termList : terms) {
                 Document doc = new Document();
                 for (String term : termList) {
-                    doc.add(new StringField("uid", term, Field.Store.NO));
+                    doc.add(new KeywordField("uid", term, Field.Store.NO));
```

Comment:
I had to change this as to avoid an `unexpected docvalues type NONE for field` error in the tests, within `SingleValueMatchQuery`. And add _some other_ objects to the `SearchExecutionContext` mock

Reply:
Right. StringField won't make doc values and we can't query them.

FWIW, I think we should make sure we fail correctly if you turn off doc values or searchability on a join key.

Reply:
Hey! That also means that `text` fields can't be joined. They just super don't have doc values.

Reply:
Just checked it, but #119473 already disallowed joins on TEXT, so it's safe for now